tests/Microsoft.Identity.Test.Common/Core/Helpers/ManagedIdentityTestUtil.cs

```diff
@@ -66,6 +66,41 @@ public static void SetEnvironmentVariables(ManagedIdentitySource managedIdentity
             }
         }
 
+        /// <summary>
+        /// Sets environment variables for testing upgrade scenarios.
+        /// This method mimics a scenario where older environment variables
+        /// (e.g., MSI_ENDPOINT and MSI_SECRET) from previous versions of
+        /// App Service (2017) still exist after an upgrade to newer versions (2019).
+        /// It ensures that MSAL's Managed Identity source detection can correctly
+        /// handle both legacy and new variables.
+        /// </summary>
+        /// <param name="managedIdentitySource">
+        /// The type of managed identity source being tested (e.g., AppService, MachineLearning).
+        /// </param>
+        /// <param name="endpoint">
+        /// The endpoint URL to be set as part of the environment variables.
+        /// </param>
+        /// <param name="secret">
+        /// Optional: The secret value to be set (default is "secret").
+        /// </param>
+        /// <param name="thumbprint">
+        /// Optional: The certificate thumbprint to be set (default is "thumbprint").
+        /// </param>
+        internal static void SetUpgradeScenarioEnvironmentVariables(ManagedIdentitySource managedIdentitySource, string endpoint, string secret = "secret", string thumbprint = "thumbprint")
```

Comment:
You don't need a separate method. Just set the env variables for managed identity too in the test with a comment that this is when app service upgrade and have the old env variables.

Reply:
> Just set the env variables for managed identity too 

not sure I understand this @neha-bhargava

Reply:
to me it is best to keep the upgrade scenario separate